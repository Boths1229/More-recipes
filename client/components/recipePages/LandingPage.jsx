import React from 'react';
import IndexHeader from '../common/IndexHeader';
import Carousel from '../Include/carousel';
import IndexMain from '../Include/IndexMain';
import IndexFooter from '../common/IndexFooter';

/**
 * @description functional component to render the landing page
 *
 * @returns {JSX} return JSX
 */
const LandingPage = () => (
  <div>
    <IndexHeader />
    <Carousel />
    <IndexMain />
    <IndexFooter />
  </div>
);


export default LandingPage;
