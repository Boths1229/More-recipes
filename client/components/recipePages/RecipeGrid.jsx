import React from 'react';
import RecipeCard from '../Include/cards/RecipeCard';

/**
 * @description functional component to render carousel
 *
 * @param {object} props
 *
 * @returns {JSX} return JSX
 */
const RecipeGrid = props => (
  props.recipeData.map(recipe => (
    <RecipeCard
      key={recipe.id}
      favRecipeData={recipe}
      recipeData={recipe}
      showActionButton={props.showActionButton}
      showRecipeDetails={props.handleShowRecipe}
      deleteRecipe={props.deleteRecipe}
      editRecipe={props.editRecipe}
      addRecipe={props.addRecipe}
      value={props.value}
      handleCloseModal={props.handleCloseModal}
      onChange={props.onChange}
      saveImageToCloud={props.saveImageToCloud}
    />
  ))
);

export default RecipeGrid;
