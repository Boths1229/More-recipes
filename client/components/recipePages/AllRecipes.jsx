import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes, { any } from 'prop-types';
import { bindActionCreators } from 'redux';
import ReactPaginate from 'react-paginate';
import { Lines } from 'react-preloading-component';
import RecipeGrid from '../recipePages/RecipeGrid';
import Header from '../common/Header';
import GuestHeader from '../common/GuestHeader';
import Footer from '../common/Footer';
import getRecipeAction from '../../actionController/getRecipe';
import getPopularRecipeAction from '../../actionController/getPopularRecipe';

/**
 * @class RecipeGrid
 *
 * @description Display all recipeCard on the app
 */
class AllRecipes extends Component {
  /**
   * @description create an instance of RecipeGrid
   *
   * @param {object} props
   */
  constructor(props) {
    super(props);
    this.state = {
      recipeData: [],
      popularRecipesData: [],
    };
    this.handlePaginationChange = this.handlePaginationChange.bind(this);
  }
  /**
   * @description call the action to display all recipes
   *
   * @param {object} props
   *
   * @return {void} call getRecipe
   */
  componentDidMount() {
    this.props.getPopularRecipeAction();
    this.props.getRecipeAction(this.props.recipes.page);
  }
  /**
   * @description update the state of recipe
   *
   * @param {object} nextProps
   *
   * @return {object} updated recipe state
   */
  componentWillReceiveProps(nextProps) {
    const { recipeData } = nextProps.recipes;
    const { popularRecipesData } = nextProps.popularRecipes;
    if (recipeData !== this.props.recipes.recipeData) {
      this.setState({
        recipeData,
      });
    }
    if (popularRecipesData !== this.props.popularRecipes.popularRecipesData) {
      this.setState({
        popularRecipesData,
      });
    }
  }

  /**
   * @description get new recipes to be displayed on the new page
   *
   * @param {object} recipes
   *
   * @return {void} calls getRecipeAction
   */
  handlePaginationChange(recipes) {
    this.props.getRecipeAction(recipes.selected);
  }

  /**
   * @description render - display all the recipes
   *
   * @return {JSX} return JSX
   */
  render() {
    const { recipeData, popularRecipesData } = this.state;
    return (
      <div>
        {
          this.props.authenticated ? <Header /> : <GuestHeader />
        }
        <div className="container pt-4">
          <div className="row">
            <section className="col-md-12">
              <h2 className="title">Popular Recipes</h2>
              <div>
                <hr />
              </div>
            </section>
            {
              !this.props.popularRecipes.isFetched ? <Lines /> : <RecipeGrid
                recipeData={popularRecipesData}
              />
            }
          </div>
          <div className="row">
            <section className="col-md-12">
              <h2 className="title">All Recipes</h2>
              <div>
                <hr />
              </div>
            </section>
          </div>
          <div className="row">
            {
              !this.props.recipes.isFetched ? <Lines /> : <RecipeGrid
                recipeData={recipeData}
              />
            }
          </div>
        </div>
        <ReactPaginate
          previousLabel="Previous"
          nextLabel="Next"
          breakLabel={<a href="">...</a>}
          breakClassName="page-link"
          onPageChange={this.handlePaginationChange}
          pageCount={this.props.recipes.page}
          containerClassName="pagination justify-content-center"
          pageLinkClassName="page-link"
          nextLinkClassName="page-link"
          previousLinkClassName="page-link"
          disabledClassName="disabled"
          pageClassName="page-item"
          previousClassName="page-item"
          nextClassName="page-item"
          activeClassName="active"
          subContainerClassName="pages pagination"
        />
        <Footer />
      </div>
    );
  }
}

AllRecipes.propTypes = {
  getRecipeAction: PropTypes.func.isRequired,
  getPopularRecipeAction: PropTypes.func.isRequired,
  recipes: PropTypes.objectOf(any).isRequired,
  authenticated: PropTypes.bool.isRequired,
  popularRecipes: PropTypes.objectOf(any).isRequired

};

/**
 * @description make state available to recipeGrid as props
 *
 * @param {object} state
 *
 * @returns {object} object
 */
const mapStateToProps = state => ({
  authenticated: state.authReducer.isAuthenticated,
  recipes: state.recipeReducer,
  popularRecipes: state.getPopularRecipeReducer
});

/**
 * @description make actions available to AllRecipes as props
 *
 * @param {function} dispatch
 *
 * @returns {void} call bindActionCreators function
 */
const mapDispatchToProps = dispatch => (
  bindActionCreators({
    getRecipeAction,
    getPopularRecipeAction
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(AllRecipes);
