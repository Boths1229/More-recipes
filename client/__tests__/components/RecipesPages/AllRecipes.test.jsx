import React from 'react';
import { shallow } from 'enzyme';
import { AllRecipes } from '../../../components/recipePages/AllRecipes';

describe('Landing page component', () => {
  it('matches snapshot', () => {
    const props = {
      getPopularRecipeAction: jest.fn(),
      getRecipeAction: jest.fn(),
      popularRecipes: {},
      recipes: {}
    };
    const output = shallow(<AllRecipes {...props} />);
    expect(output).toMatchSnapshot();
  });
  describe('instances and life cycle methods', () => {
    const props = {
      popularRecipes: {},
      recipes: {},
      getRecipeAction: jest.fn(),
      getPopularRecipeAction: jest.fn()
    };
    it('handlePaginationChanges', () => {
      const wrapper = shallow(<AllRecipes {...props} />);
      wrapper.instance().handlePaginationChange({ recipes: {} });
      expect(props.getRecipeAction).toBeCalled();
    });
  });
});
