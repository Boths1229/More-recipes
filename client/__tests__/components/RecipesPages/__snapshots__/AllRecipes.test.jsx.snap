// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page component matches snapshot 1`] = `
<div>
  <GuestHeader />
  <div
    className="container pt-4"
  >
    <div
      className="row"
    >
      <section
        className="col-md-12"
      >
        <h2
          className="title"
        >
          Popular Recipes
        </h2>
        <div>
          <hr />
        </div>
      </section>
      <n
        color="#FF4601"
        size={30}
      />
    </div>
    <div
      className="row"
    >
      <section
        className="col-md-12"
      >
        <h2
          className="title"
        >
          All Recipes
        </h2>
        <div>
          <hr />
        </div>
      </section>
    </div>
    <div
      className="row"
    >
      <n
        color="#FF4601"
        size={30}
      />
    </div>
  </div>
  <PaginationBoxView
    activeClassName="active"
    breakClassName="page-link"
    breakLabel={
      <a
        href=""
      >
        ...
      </a>
    }
    containerClassName="pagination justify-content-center"
    disableInitialCallback={false}
    disabledClassName="disabled"
    marginPagesDisplayed={3}
    nextClassName="page-item"
    nextLabel="Next"
    nextLinkClassName="page-link"
    onPageChange={[Function]}
    pageClassName="page-item"
    pageCount={10}
    pageLinkClassName="page-link"
    pageRangeDisplayed={2}
    previousClassName="page-item"
    previousLabel="Previous"
    previousLinkClassName="page-link"
    subContainerClassName="pages pagination"
  />
  <Footer />
</div>
`;
